File: sample/use_cocoapods/objectivec/objectivec/FullscreenInterstitial/FullscreenInterstitialViewController.m
#import "FullscreenInterstitialViewController.h"

@interface FullscreenInterstitialViewController ()

@end

@implementation FullscreenInterstitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _zoneIDText.delegate = self;
    
    [GNSFullscreenInterstitialAd sharedInstance].delegate = self;
    
    self.showAdButton.enabled = NO;
}

- (void)requestFullscreenInterstitialAd {
    self.loadAdButton.enabled = NO;
    GNSRequest *request = [GNSRequest request];
    //request.geoLocationEnable = NO;
    request.GNAdlogPriority = GNLogPriorityInfo;
    [[GNSFullscreenInterstitialAd sharedInstance] loadRequest:request withZoneID:_zoneIDText.text];
}

- (IBAction)loadInterstitialAd:(id)sender {
    [self requestFullscreenInterstitialAd];
}

- (IBAction)showInterstitialAd:(id)sender {
    if ([[GNSFullscreenInterstitialAd sharedInstance] canShow]) {
        [[GNSFullscreenInterstitialAd sharedInstance] show:self];
    }
}

// Hide the keyboard when press return key in a UITextField
-(BOOL) textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    return YES;
}

#pragma mark GNSFullscreenInterstitialAdDelegate
- (void)fullscreenInterstitialDidReceiveAd:(GNSFullscreenInterstitialAd *)fullscreenInterstitial
{
    NSLog(@"FullscreenInterstitialAd received.");
    self.showAdButton.enabled = YES;
}

- (void)fullscreenInterstitial:(GNSFullscreenInterstitialAd *)fullscreenInterstitial didFailToLoadWithError:(NSError *)error
{
    NSLog(@"FullscreenInterstitialAd load failed: %@", error.description);
    self.loadAdButton.enabled = YES;
}

- (void)fullscreenInterstitialAdDidClick:(GNSFullscreenInterstitialAd *)fullscreenInterstitial
{
    NSLog(@"FullscreenInterstitialAd clicked.");
}

- (void)fullscreenInterstitialAdDidClose:(GNSFullscreenInterstitialAd *)fullscreenInterstitial
{
    NSLog(@"FullscreenInterstitialAd Closed");
    self.loadAdButton.enabled = YES;
    self.showAdButton.enabled = NO;
}

- (void)fullscreenInterstitialWillPresentScreen:(GNSFullscreenInterstitialAd *)fullscreenInterstitial
{
    NSLog(@"FullscreenInterstitialAd did presented");
}


@end

File: sample/use_cocoapods/swift/swift/FullscreenInterstitial/FullscreenInterstitialViewController.swift
import UIKit

class FullscreenInterstitialViewController: UIViewController, UITextFieldDelegate, GNSFullscreenInterstitialAdDelegate {
    
    @IBOutlet weak var loadAdButton: UIButton!
    @IBOutlet weak var showAdButton: UIButton!
    @IBOutlet weak var zoneIdText: UITextField!
    @IBOutlet weak var errorLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        zoneIdText.delegate = self
        GNSFullscreenInterstitialAd.sharedInstance().delegate = self
        self.showAdButton.isEnabled = false
        errorLabel.text = ""
    }
    
    func requestInterstitialAd() {
        self.loadAdButton.isEnabled = false
        let request = GNSRequest()
        // set log level
        request.gnAdlogPriority = GNLogPriorityInfo
        if let idStr = zoneIdText.text {
            if idStr.isEmpty {
                NSLog("Please input zone id.")
                showAlert(title: "Please input zone id.",
                          message: "Required zone id to request ad.")
            }
            GNSFullscreenInterstitialAd.sharedInstance().load(request, withZoneID: idStr)
        }
    }

    @IBAction func loadInterstitialAd(_ sender: Any) {
        self.requestInterstitialAd()
    }
    
    @IBAction func showInterstitialAd(_ sender: Any) {
        if(GNSFullscreenInterstitialAd.sharedInstance().canShow()) {
            GNSFullscreenInterstitialAd.sharedInstance().show(self)
        } else {
            showAlert(title: "Fullscreen interstitial not ready",
                      message: "The Fullscreen interstitial didn't finish loading or failed to load or need to reload")
        }
    }
    
    // MARK GNSFullscreenInterstitialAdDelegate
    func fullscreenInterstitialDidReceive(_ fullscreenInterstitial: GNSFullscreenInterstitialAd!) {
        NSLog("ViewController: Fullscreen ad is received.")
        self.showAdButton.isEnabled = true
        errorLabel.text = ""
    }
    
    func fullscreenInterstitial(_ fullscreenInterstitial: GNSFullscreenInterstitialAd!, didFailToLoadWithError error: Error!) {
        NSLog("ViewController: Fullscreen ad failed to load. error: %@", error.localizedDescription)
        self.loadAdButton.isEnabled = true
        errorLabel.text = error.localizedDescription
    }
    
    func fullscreenInterstitialAdDidClose(_ fullscreenInterstitial: GNSFullscreenInterstitialAd!) {
        NSLog("ViewController: Fullscreen ad is closed.")
        self.loadAdButton.isEnabled = true
        self.showAdButton.isEnabled = false
    }
    
    func fullscreenInterstitialAdDidClick(_ fullscreenInterstitial: GNSFullscreenInterstitialAd!) {
        NSLog("ViewController: Fullscreen ad click.")
    }
    
    func fullscreenInterstitialWillPresentScreen(_ fullscreenInterstitial: GNSFullscreenInterstitialAd!) {
        NSLog("ViewController: Fullscreen ad showing.")
    }
    
    
    //MARK UITextFieldDelegate
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        // Hide the keyboard when press return key in a UITextField
        textField.resignFirstResponder()
        return true
    }
    
    func showAlert(title:String, message:String) {
        let alertController = UIAlertController(title: title,
                                                message: message,
                                                preferredStyle: UIAlertController.Style.alert)
        alertController.addAction(UIAlertAction(title: "OK",
                                                style: UIAlertAction.Style.cancel,
                                                handler:nil))
        present(alertController, animated: true, completion: nil)
    }
}

